buildenv/jenkins/common/build.groovy

```diff
@@ -155,8 +155,19 @@ def checkout_pullrequest() {
 
         // Setup PR_IDs for dependent changes
         for (DEPEND in DEPENDS_ARRAY) {
-            String REPO = DEPEND.substring(DEPEND.indexOf("/") + 1, DEPEND.indexOf("#"));
-            String PR_ID = DEPEND.substring(DEPEND.indexOf("#") + 1, DEPEND.length());
+            switch (DEPEND) {
+                case ~/https:\/\/github\.(ibm\.)?com\/\S+\/\S+\/pull\/[0-9]+/:
```

Comment:
I like the idea here, but the pattern is a little permissive: `\S` will match `/` so there can be any number (>= 3) of slashes between `com` and `pull`. The same issue exists in the case on line 164. Replacing each `\S` with `[^\s/]` should fix that.